portality/store.py

```diff
@@ -112,6 +112,11 @@ def delete(self, container_id, target_name=None):
         # we are not allowed to delete the bucket, so we just delete the contents
         keys = self.list(container_id)
 
+        if target_name is not None:
```

Comment:
Ahah, I now see where we were deleting the entire contents of the bucket!

I don't really like this delete function yet, though - it makes sense to check whether the delete target object is in the bucket before we issue a delete request via the client, but I think the default behaviour of `target_name=None` is potentially dangerous since it will still clear the bucket. If that's the intention, it should at least be noted in a function docstring!

Reply:
Yes, that is the intention - you can use this method to delete a specific file, or the entire bucket.  What would you do instead?  The obvious alternative is to separate delete of a target from delete of a bucket at the method level.  If we want to make that change, now is a good time!

So, for example

```
store.delete_container(container_id)
store.delete_file(container_id, target_name)
```

Reply:
I have gone ahead and done this, just doing some more testing to make sure everything works as expected, will update this PR shortly.

Reply:
Fantastic, that's pretty much what I had in mind - I think it's sensible to keep that clear and separate. Documenting the behaviour would've been enough, though!